### PR TITLE
fix: address clippy warnings

### PR DESCRIPTION
**Pull request by yaahallo**
This commit is all about making clippy happy, and to a lesser extent, Jane.

### CODE REVIEW

**Comment by davidbarsky** [comment]
@yaahallo: Thanks for your help! I think we can safety bump up the minimum rustc version from 1.26 to 1.34, if @hawkw is okay with that?

**Comment by hawkw** [comment]
@davidbarsky I would strongly prefer that we continue to track `tokio`'s minimum Rust version for everything except for the proc-macro crates, though I'm open to negotiation.

File: tokio-trace-futures/examples/proxy_server.rs [hunk 1]
```diff
@@ -23,10 +23,14 @@ use tokio::net::{TcpListener, TcpStream};
...
-    let listen_addr = env::args().nth(1).unwrap_or("127.0.0.1:8081".to_string());
+    let listen_addr = env::args()
+        .nth(1)
+        .unwrap_or_else(|| "127.0.0.1:8081".to_string());
     let listen_addr = listen_addr.parse::<SocketAddr>()?;
 
-    let server_addr = env::args().nth(2).unwrap_or("127.0.0.1:3000".to_string());
+    let server_addr = env::args()
+        .nth(2)
+        .unwrap_or_else(|| "127.0.0.1:3000".to_string());
```
Comment (hawkw) [inline tokio-trace-futures/examples/proxy_server.rs]: Out of curiosity: is this a rustfmt suggestion or a clippy suggestion?

File: tokio-trace-subscriber/src/registry.rs [hunk 2]
```diff
@@ -171,17 +171,15 @@ impl RegisterSpan for IncreasingCounter {
 
     fn new_span(&self, _new_span: &Attributes) -> Id {
         let id = self.next_id.fetch_add(1, Ordering::SeqCst);
-        let id = Id::from_u64(id as u64);
+        Id::from_u64(id as u64)
         // if let Ok(spans) = self.spans.lock() {
```
Comment (hawkw) [inline tokio-trace-subscriber/src/registry.rs]: the commented out code should probably also be removed; this is why we had the (now unnecessary) let binding...

File: tokio-trace-fmt/src/span.rs [hunk 3]
```diff
@@ -472,14 +472,14 @@ impl Slot {
 impl Slab {
     #[inline]
     fn write_slot(&self, idx: usize) -> Option<RwLockWriteGuard<Slot>> {
-        self.slab.get(idx).map(|slot| slot.write())
+        self.slab.get(idx).map(RwLock::write)
     }
 
     #[inline]
-    fn read_slot<'a>(&'a self, idx: usize) -> Option<RwLockReadGuard<'a, Slot>> {
```
Comment (hawkw) [inline tokio-trace-fmt/src/span.rs]: is this a semantic difference?

**Comment by hawkw** [comment]
@davidbarsky also, if we do bump the minimum Rust version, we shouldn't bump it to 1.34; IIRC tokio's policy is that we should support at least the last six stable releases.

File: tokio-trace-futures/examples/proxy_server.rs (hunk 1 above)
Comment (yaahallo) [reply tokio-trace-futures/examples/proxy_server.rs]: looks like rustfmt, I have it setup to run on save so its pretty easy for it to sneak in reformats :S

File: tokio-trace-fmt/src/span.rs (hunk 3 above)
Comment (yaahallo) [reply tokio-trace-fmt/src/span.rs]: Based on my understanding of lifetime ellision rules, no.

I remember from before though you mentioned `'_` not being supported in some versions, so I need to double check that this compiles in 1.26 locally.

File: tokio-trace-futures/examples/proxy_server.rs (hunk 1 above)
Comment (LucioFranco) [reply tokio-trace-futures/examples/proxy_server.rs]: I think this is a clippy one so that you only run to_string one the unwrap on an err iirc.

Comment (yaahallo) [reply tokio-trace-futures/examples/proxy_server.rs]: That's exactly what it was, good catch.

File: tokio-trace-fmt/src/span.rs (hunk 3 above)
Comment (hawkw) [reply tokio-trace-fmt/src/span.rs]: I can say right now that this won't compile on 1.26. I'd be fine with adding the clippy ignore attribute for that lint for now?

File: tokio-trace-futures/examples/proxy_server.rs (hunk 1 above)
Comment (hawkw) [reply tokio-trace-futures/examples/proxy_server.rs]: Oh I missed that, thanks.

FWIW the formatting changes are fine regardless, I was just curious.

**Comment by hawkw** [comment]
Nevermind, I've been informed that `'_` was stabilized in 1.26. So this is fine.

**Pull request merged**